### PR TITLE
Fix spelling and word choice issues

### LINKED ISSUE #914: Spelling and word choice issues (mix of British and American strings in strings.xml)
This is just a spin-off of the discussion in #883. Currently, the `strings.xml` file has a mix of British and American strings, with a few spelling mistakes and typographic issues. Here we can discuss how to handle this.

I think US English is *usually* used as the default for English-based Android apps, with UK English provided as a translation. But it looks this isn’t a requirement, and in #883 there was a preference (?) for having UK English as the default. So here’s my proposal:

My proposal:
- The US English `strings.xml` will be changed to UK English. At the same time, any spelling mistakes found will be fixed (e.g. ‘housenumber’ will be changed to ‘house number’) and typographic fixes will be applied (e.g. hyphens used as dashes will be changed to typographically correct en-dashes (- to –) and ASCII apostrophes will be changed to typographically correct apostrophes (' to ’). I can create a PR for this.
- A US English translation is added to POEditor. This will for example change ‘cancelled’ into ‘canceled’ and ‘car park’ into ‘parking lot’. I can create the initial translation, but it would be nice if someone could proof-read it.

Even for UK English, there are spelling and grammar choices to be made. Based on the style used in the official Android translation (and in other free software projects), I propose to use -ise instead of -ize and to omit the listing comma in front of ‘and’/‘or’ (e.g. ‘red, green and blue’, not ‘red, green, and blue’). (There might be other relevant style choices I have missed, but I will mention them in this issue if I discover any when reviewing the strings.)

Comments?

### PR DESCRIPTION
**Pull request by huftis**
Basically, this fixes #914 (see that issue for a complete discussion). Here’s a short summary.

- Fixes spelling mistakes (e.g. ‘housenumber’ → ‘house number’), grammar mistakes and inconsistent use of lowercase/uppercase words.
- Changes strings to use typographically correct characters (e.g. real en-dashes and apostrophes).
- The strings were a mix of British and US spelling. This PR changes all strings to use US spelling, word choices and punctuation.
- Adds a British translation (locale code: en-rGB). (This basically reverts most of the above spelling / word choice / punctuation changes.)

### CODE REVIEW

**Comment by HolgerJeromin** [comment]
Is `/values-en-rGB/strings.xml` a typo? 
I would expect `/values-en-GB/strings.xml`

**Comment by huftis** [comment]
@HolgerJeromin No, it’s not a typo. For some reason, Android uses these non-standard locale codes. https://developer.android.com/guide/topics/resources/providing-resources#LocaleQualifier:

> The language is defined by a two-letter ISO 639-1 language code, optionally followed by a two letter ISO 3166-1-alpha-2 region code (preceded by lowercase r).
>
> The codes are not case-sensitive; the r prefix is used to distinguish the region portion. You cannot specify a region alone.

**Comment by HolgerJeromin** [comment]
Thanks. 
Seems odd but is a good idea to prevent mixing the region with the locale.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -255,65 +254,65 @@ The info you enter is then directly added to the OpenStreetMap in your name, wit
...
+    <string name="quest_streetName_menuItem_language_simple">%1$s – %2$s</string>
+    <string name="quest_streetName_menuItem_language_native">%1$s – %2$s (%3$s)</string>
+    <string name="quest_streetName_answer_cantType">"Can’t type the characters on the sign"</string>
     <string name="quest_streetName_cantType_title">Missing keyboard layout?</string>
     <string name="quest_streetName_cantType_open_settings">Open settings</string>
     <string name="quest_streetName_cantType_open_store">Visit app store</string>
     <string name="quest_streetName_cantType_description">"If you use a default keyboard, chances are that you can simply enable the keyboard layout for the language you need in the input method settings.
 
-Otherwise, you can download another keyboard in the app store. Popular keyboards that support many languages are Google Gboard, SwiftKey Keyboard and Multiling O Keyboard."</string>
+Otherwise, you can download another keyboard in the app store. Popular keyboards that support many languages are Google Gboard, SwiftKey Keyboard, and Multiling O Keyboard."</string>
```
Comment (westnordost) [inline app/src/main/res/values/strings.xml]: Sure about this?

Comment (huftis) [reply app/src/main/res/values/strings.xml]: Yes, US English use the listing comma (I’ve removed it in the en-GB translation).

Comment (westnordost) [reply app/src/main/res/values/strings.xml]: OK, merging!

**Comment by westnordost** [comment]
Also imported to POEditor, changed English to US-English and cleared all previous contributors so that they can reassign themselves to either GB or US

**Pull request merged**